### PR TITLE
Determine the minimum gradle version based on the wrapper

### PR DESCRIPTION
**Pull request by alpar-t**
This is restrictive and forces users of the plugin to move together with
us, but without integration tests it's close to impossible to make sure
that the claimed compatability is really there.
If we do want to offer more flexibility, we should add those tests
first.

@nik9000 this is the promised follow-up to #32200

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -78,6 +83,7 @@ task writeVersionProperties {
 processResources {
   dependsOn writeVersionProperties
   from tempPropertiesFile
+  from wrapperProperties
```
Comment (nik9000) [inline buildSrc/build.gradle]: I think we should rename this file so it is more obvious what we are using it for. Or maybe not write the entire file, just the version.

**Review by nik9000 — COMMENTED** [review]
Do you think it is worth grabbing the version from the zip file in the distribution url so we don't have to write more stuff?

File: buildSrc/build.gradle (hunk 1 above)
Comment (atorok) [reply buildSrc/build.gradle]: I initially added the file as is thinking about parsing it from the URL, but then realized that the wrapper know about the version it's going to use. In retrospect I'm not sure why I choose to add it to the properties files, I think the initial idea rooted, it makes much more sense to add a `minimumGradleVersion` resource that just has this version number.

File: build.gradle [hunk 2]
```diff
@@ -537,15 +533,19 @@ task run(type: Run) {
...
+    distributionType = 'ALL'
     doLast {
         final DistributionLocator locator = new DistributionLocator()
         final GradleVersion version = GradleVersion.version(wrapper.gradleVersion)
         final URI distributionUri = locator.getDistributionFor(version, wrapper.distributionType.name().toLowerCase(Locale.ENGLISH))
         final URI sha256Uri = new URI(distributionUri.toString() + ".sha256")
         final String sha256Sum = new String(sha256Uri.toURL().bytes)
         wrapper.getPropertiesFile() << "distributionSha256Sum=${sha256Sum}\n"
-        println "Added checksum to wrapper properties"
+        println "Added checksum and version to wrapper properties"
```
Comment (nik9000) [inline build.gradle]: I think this println isn't accurate any more, right?

**Comment by nik9000** [comment]
> My hope is that we'll be able to add some tests soon so that when we move to Gradle 4.10 we can add a test for 4.9 for confidence that we can still support it, then we might stop auto updating it with the wrapper.

I think we have so much on our plate right now that that isn't worth it.

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -4,3 +4,4 @@ distributionUrl=https\://services.gradle.org/distributions/gradle-4.9-all.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
 distributionSha256Sum=39e2d5803bbd5eaf6c8efe07067b0e5a00235e8c71318642b2ed262920b27721
+gradleVersion=4.9
```
Comment (nik9000) [inline gradle/wrapper/gradle-wrapper.properties]: This won't be in the file if you regenerate it, right?

**Review by nik9000 — CHANGES_REQUESTED** [review]

**Review by nik9000 — APPROVED** [review]

**Comment by nik9000** [comment]
LGTM!

**Pull request merged**